runbotics/runbotics-desktop/src/core/websocket/listeners/process.listener.ts

```
@@ -51,6 +51,14 @@ export class ProcessListener {
     async startProcess(data: StartProcessMessageBody) {
         this.logger.log(`=> Incoming message to start process (id: ${data.processId})`);
 
+        const { processInstancesCount } = this.runtimeService.checkRuntimeStatus();
+
+        if (processInstancesCount > 0) {
+            this.logger.log('<= Process is already running');
+            this.io.emit(BotWsMessage.RUNNING_PROCESS, BotStatus.BUSY);
+            return;
```

you should throw an error here to end execution of a process